### PR TITLE
feat: added option to enable image cleaner to container service aks. - `avm/res/container-service/managed-cluster`

### PR DESCRIPTION
**Pull request by PixelRobots**
## Description

This gives you user the option to enable image cleaner on an AKS cluster

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.container-service.managed-cluster](https://github.com/PixelRobots/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)       |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/container-service/managed-cluster/main.bicep [hunk 1]
```diff
@@ -297,6 +297,12 @@ param enableWorkloadIdentity bool = false
 @description('Optional. Whether to enable Azure Defender.')
 param enableAzureDefender bool = false
 
+@description('Optional. Whether to enable Image Cleaner for Kubernetes.')
+param enableImageCleaner bool = false
+
+@description('Optional. The interval in hours Image Cleaner will run. Defaults 120, 5 days.')
+param imageCleanerIntervalHours int = 120
```
Comment (ilhaan) [inline avm/res/container-service/managed-cluster/main.bicep:304]: The default interval hours when using Azure CLI is one week (168 hours). The minimum is 24 hours. 

I think we should set this to 24 hours by default. 

https://learn.microsoft.com/en-us/azure/aks/image-cleaner#configuration-options

CC: @JPEasier

**Review by ilhaan — CHANGES_REQUESTED** [review]

File: avm/res/container-service/managed-cluster/main.bicep (hunk 1 above)
Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep:304]: Happy to change it if needed. I just used the value I use internally. I found it a good fit.

Comment (ilhaan) [inline avm/res/container-service/managed-cluster/main.bicep:304]:
```suggestion
@description('Optional. The interval in hours Image Cleaner will run. Minimum value is 24 hours and maximum is three months')
param imageCleanerIntervalHours int = 24
```

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep:304]: Thanks. I will get this sorted first thing in the morning and then kick off another update and test. 

I have a few more to do soon too. It's becoming easier now. Thanks to all of your help.

**Review by ilhaan — DISMISSED** [review]

**Comment by ilhaan** [comment]
@PixelRobots conflicts need fixing before we can merge.

**Comment by PixelRobots** [comment]
That should be fixed now.

File: avm/res/container-service/managed-cluster/main.bicep [hunk 2]
```diff
@@ -304,6 +304,12 @@ param enableWorkloadIdentity bool = false
 @description('Optional. Whether to enable Azure Defender.')
 param enableAzureDefender bool = false
 
+@description('Optional. Whether to enable Image Cleaner for Kubernetes.')
+param enableImageCleaner bool = false
+
+@description('Optional. The interval in hours Image Cleaner will run. Minimum value is 24 hours and maximum is three months.')
+param imageCleanerIntervalHours int = 24
```
Comment (AlexanderSehr) [inline avm/res/container-service/managed-cluster/main.bicep:311]:
```suggestion
@description('Optional. The interval in hours Image Cleaner will run. The maximum value is three months.')
@minValue(24)
param imageCleanerIntervalHours int = 24
```

Comment (AlexanderSehr) [reply avm/res/container-service/managed-cluster/main.bicep:311]: Woud have liked to add a maxValue as well, but `3 months` is a bit ... odd to calculate as not every month has the same length. I guess, `3 x 31 days x 24 hours = 2232` would at least set some limit, but I'm not sure if it's a helpful limit.

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep]: Makes sense to me.

File: avm/res/container-service/managed-cluster/main.json [hunk 3]
```diff
@@ -1270,6 +1270,20 @@
...
       }
     },
+    "enableImageCleaner": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Optional. Whether to enable Image Cleaner for Kubernetes."
+      }
+    },
+    "imageCleanerIntervalHours": {
```
Comment (AlexanderSehr) [inline avm/res/container-service/managed-cluster/main.json:1280]: Must be refreshed

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.json:1280]: I will get this updated tomorrow when I get back on my laptop. Thanks for being patient.

Comment (AlexanderSehr) [reply avm/res/container-service/managed-cluster/main.json:1280]: All good. Thanks for putting in the work 💪

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.json:1280]: sset-avmmodules has been reran and changes pushed. this should be ready to go now.

**Comment by PixelRobots** [comment]
Any update on this one being merged?

**Comment by AlexanderSehr** [comment]
> Any update on this one being merged?

cc: @ilhaan / @JPEasier

**Review by JPEasier — APPROVED** [review]

**Pull request merged**